### PR TITLE
Adds Deployment Version to the Flow Run object

### PR DESCRIPTION
**Pull request by masonmenges**
This PR adds the deployment version to flow run objects for better traceability/lineage in determining which version of code ran when a flow run was triggered. 

### Example
```python
@flow
async def demo_flow(date: datetime = None):
    logger = get_run_logger()
    context = get_run_context()    
    flow_run_id = context.flow_run.id
    deployment_id = context.flow_run.deployment_id

    async with get_client() as client:
        dep = await client.read_deployment(deployment_id)
        flow_run = await client.read_flow_run(flow_run_id)
        logger.info("deployment version from Deployment: " + dep.version)
        logger.info("Deployment version from Flow Run: " + flow_run.deployment_version)
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

closes https://github.com/PrefectHQ/prefect/issues/11983

### CODE REVIEW

**Review by chrisguidry — APPROVED** [review]

File: src/prefect/server/models/deployments.py [hunk 1]
```diff
@@ -619,6 +619,7 @@ async def _generate_scheduled_flow_runs(
                     "id": uuid4(),
                     "flow_id": deployment.flow_id,
                     "deployment_id": deployment_id,
+                    "deplyment_version": deployment.version,
```
Comment (meggers) [inline src/prefect/server/models/deployments.py:622]: deplyment_version is this an issue?

Comment (masonmenges) [reply src/prefect/server/models/deployments.py:622]: @meggers 
> deplyment_version is this an issue?

It shouldn't be no, but for clarification are you just referring to the making convention?

Comment (meggers) [reply src/prefect/server/models/deployments.py:622]: Sorry for my lack of clarity, was referring to the misspelling (missing o in deployment)

Comment (masonmenges) [reply src/prefect/server/models/deployments.py:622]: Ah yep thanks for flagging that I'll get it updated 😁

**Pull request merged**